spec.html

```diff
@@ -8274,7 +8274,14 @@
         <h1>[[DefineOwnProperty]] ( _P_, _Desc_ )</h1>
         <p>When the [[DefineOwnProperty]] internal method of a module namespace exotic object _O_ is called with property key _P_ and Property Descriptor _Desc_, the following steps are taken:</p>
         <emu-alg>
-          1. Return *false*.
+          1. Let _current_ be ? _O_.[[GetOwnProperty]](_P_).
```

Comment:
The first step needs to be `If Type(P) is Symbol, return OrdinaryDefineOwnProperty(O, P, Desc).`, because the symbol properties of module namespace objects have different property descriptor attributes compared to the export properties.

Reply:
Meh, I just noticed there is an issue when the current definition of OrdinaryDefineOwnProperty is directly called:
`OrdinaryDefineOwnProperty(...)` directly accesses the [[Extensible]] internal slot instead of calling [[IsExtensible]], and module namespace objects don't have an [[Extensible]] internal slot.

So we've got two choices:
- Either change `OrdinaryDefineOwnProperty(...)` to call [[IsExtensible]] instead of directly accessing [[Extensible]].
- Or add an [[Extensible]] internal slot to module namespace objects.

Reply:
I thing the first alternative is better.  Providing alternative essential internal method definitions was intended to be the manner to  encode such values that apply to all instance of a specific kind of exotic object. 

I dislike the direction the second alternative is going. The [[Extensible]] internal slot is one of the defining characteristics of an ordinary object. Namespace objects are about as unordinary as you can get. Adding slots simply to allow use of Ordinary* internal methods on such non-ordinary objects weakens the concept of an ordinary object within the spec.

In ES6, Module namespace objects originally did not have a [[Prototype]] slot for similar reasons. Instead, it had a [[GetPrototypeOf]] internal method that always returned null. For some reason, this has changed to eliminate eliminate the exotic [[GetPrototypeOf]] and add a [[Prototype]] slot that always "contains the value null".  I suspect this was done for a similar reason to choice 2 above. That change should probably be reverted.

In both cases, these sounds like changes that were simply made to facilitate pseudo-code reuse, but I think such changes detract rather than add to the clarity of the spec.  Internal slots are instance specific, stateful and generally mutable.   Their presence should imply at the very least that the state may differ among different instances of that kind of exotic object.  When a value is immutable and constant that is no need to have instance specific slot to hold the constant value. The better way is to provide exotic internal methods (such as [[isExtensible]] or [[GetPrototypeOf]]) that directly use the constant.  Such internal methods are conceptually shared among of instance of the exotic object type.